activesupport/lib/active_support/duration/iso8601_serializer.rb

```diff
@@ -19,8 +21,8 @@ def serialize
         output = +"P"
         output << "#{parts[:years]}Y"   if parts.key?(:years)
         output << "#{parts[:months]}M"  if parts.key?(:months)
-        output << "#{parts[:weeks]}W"   if parts.key?(:weeks)
         output << "#{parts[:days]}D"    if parts.key?(:days)
+        output << "#{parts[:weeks]}W"   if parts.key?(:weeks) && !week_mixed_with_date?(parts)
```

Comment:
Why do we need `&& !week_mixed_with_date?(parts)` here if normalize is always ran before this, removing weeks?